### DISCUSSION TITLE
How is uBlock released to the browser stores?

### DISCUSSION THREAD

**Discussion by thinkswan** [uBlockOrigin/uBlock-issues#2148] [General] [answered]
Hi! 👋 I'm a longtime fan of uBlock Origin and, considering the tens of millions of people who rely on it, am curious about how it's released to the browser stores. I read through the wiki and past discussions, but couldn't find any documentation about it.

How is uBlock released to the browser stores in a safe way? How are rollbacks handled in case a major bug makes it live? Are [partial rollouts](https://developer.chrome.com/docs/webstore/update/#partial-rollout) used?

**Comment by u-RraaLL** [comment]
Some of your questions are answered here: https://www.reddit.com/r/uBlockOrigin/comments/vavtgj/ublock_origin_143_announcement_thread/

I haven't seen a rollback yet. Usually if a major issue is detected in a stable release, there's an emergency fix pushed out asap.

**Reply by thinkswan** [reply]
Thanks for linking that thread! It sounds like partial rollout is used for Chrome (a few percent each day) and submissions to other stores occur when a bug-free version is live and verified.

I'm still curious about how the concept of a rollback is handled, since the browser stores don't actually support this feature. If the release process is documented anywhere, that would be golden. 🤩

**Reply by gorhill** [reply, marked as answer]
The web stores don't allow rollback -- I just published a new revision with emergency fixes when there are serious issues. I trust enough people are using the dev build to catch serious issues before publishing to all.

**Reply by thinkswan** [reply]
Ah, so the dev build is where serious issues are sniffed out before release, makes sense!
